docs/components/receive.md

```diff
@@ -77,6 +77,25 @@ The example content of `hashring.json`:
 
 With such configuration any receive listens for remote write on `<ip>10908/api/v1/receive` and will forward to correct one in hashring if needed for tenancy and replication.
 
+## Limiting
+
+### Request limits (work in progress)
+
+Thanos Receive supports setting limits on the incoming remote write request sizes. These limits should help you to prevent a single tenant from being able to send big requests and possibly crash the Receive.
+
+These limits are applied per request and can be configured with the following command line arguments:
+
+- `--receive.write-request-limits.max-size-bytes`: the maximum body size.
+- `--receive.write-request-limits.max-concurrency`: the maximum amount of remote write requests that will be concurrently processed while others wait.
+- `--receive.write-request-limits.max-series`: the maximum amount of series in a single remote write request.
+- `--receive.write-request-limits.max-samples`: the maximum amount of samples in a single remote write request (summed from all series).
+
+Any request above these limits will cause an 413 HTTP response (*Entity Too Large*) and should not be retried without modifications. It's up to remote write clients to split up the data and retry or completely drop it.
```

Comment:
Agreed. I'll write more about this. 👍 

> I'm wondering however whether there'd be a will to do something like that in the upstream 🤔, to process 413 in some way other than just dropping it.

That's something I've considered. I think we would have to add error codes to the response, so that a client (Prometheus or something else) could know whether the 413 is because of too many series, too many samples, or simply request body size, and react accordingly so that however it splits the request it'll be accepted next time (or likely accepted).